lib/LanguageServer/Handler/PhpactorHandlerLoader.php

```diff
@@ -56,7 +58,9 @@ protected function createContainer(string $rootUri, array $config): Container
 
         $container = PhpactorContainer::fromExtensions(
             $extensionClasses,
-            array_merge($parameters, $config)
+            array_merge($parameters, $params->initializationOptions, [
+                LanguageServerExtension::PARAM_CLIENT_CAPABILITIES => $params->capabilities
```

Comment:
:+1: (i think)

Reply:
> they are here able to add the capabilities in the config

You mean `initializationOptions` ?
If so then I just have to inverse the order no ?

And I personally understoof `initializationOptions` as being options related to the server not the client, but the spec is light about that.

Reply:
I just realized, should not I use `array_merge_recursive` instead ?
Since the capabilities are multiple dimensional array.
Or we need to "normalize" it in order to recreate keys like `textDocument.completion`.
Which I think I will do anyway in a new service `ClientCapabilitiesProvider`

The main reason for that service is that I don't like to handle array for starter, there is no way to define a contract about what is in it.
Second we are forced to either know it's structure in the container or provide it as a hole to a service requiring one of the client capabilities.
I feel like the container should not be the one knowing which capabilities to inject in a service, if we really want to optimize our services by providing only the relevant capabilities instead of all of them then we should use a dedicated factory.